tests/test_unit_thread_pool.cpp

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2018 Snowflake Computing, Inc. All rights reserved.
+ */
+
+#include "utils/test_setup.h"
+#include "util/ThreadPool.hpp"
+#include <thread>
+
+void test_thread_pool(void **unused)
+{
+  Snowflake::Client::Util::ThreadPool tp(4);
+
+  for (int i=0; i<10; i++)
+  {
+    tp.AddJob([=]{
+      std::this_thread::sleep_for(std::chrono::seconds(4));
```

Comment:
Is there any way to verify all of the threads are completed?

Reply:
The thread pool I implemented cannot return future. (It requires some additional work to implement future)

I can verify by passing in a variable reference declared in main thread and update the value in worker thread. After threads are done, check those values are updated or not.